src/validation.cpp

```diff
@@ -1858,12 +1858,58 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
     // before the first had been spent.  Since those coinbases are sufficiently buried its no longer possible to create further
     // duplicate transactions descending from the known pairs either.
     // If we're on the known chain at height greater than where BIP34 activated, we can save the db accesses needed for the BIP30 check.
+
+    // The exception to this logic is if there are coinbases from blocks before
+    // the BIP34 height of 227,931 which have a scriptSig with a valid-looking
+    // BIP34 height for a later block height. These coinbases run the risk of
+    // being duplicated when we reach that later block height. An exhaustive
+    // search of all mainnet coinbases before the BIP34 height which have a
+    // scriptSig indicating a later block height reveals 2 cases where the later
+    // block height is before 1,983,702.
+
+    // There are many cases which indicate block height 1,983,702 or later, so
+    // we simply remove the optimiztion to skip BIP30 checking for blocks at
+    // height 1,983,702 or higher.  Before we reach that block in another 25
+    // years or so, we should take advantage of a future consensus change to do
+    // a new and improved version of BIP34 that will actually prevent ever
+    // creating any duplicate coinbases in the future.
+
+    // The first earlier case is block 209,920 and the indicated height is
+    // 209,921, so we do not need to be concerned with this case because 209,921
+    // is under the BIP34 height and so BIP30 is active already.
+
+    // The second case is block 176,684 which has an indicated block height of
+    // 490,897. Unfortunately this issue was not discovered until about 2 weeks
+    // before block 490,897 so there was not much opportunity to address this
+    // case other than to carefully analyze it and determine it would not be a
+    // problem. Block 490,897 was, in fact, mined with a different coinbase
+    // than 176,684, but it is important to note that even if it hadn't
```

Comment:
suggested rephrase: "than block 176,684's"

I was reading it as the the height in coinbase.